src/poetry/repositories/legacy_repository.py

```diff
@@ -45,14 +47,13 @@ def find_packages(self, dependency: Dependency) -> list[Package]:
 
         ignored_pre_release_versions = []
 
-        if self._cache.store("matches").has(key):
-            versions = self._cache.store("matches").get(key)
-        else:
+        versions = self._match_cache.get(key)
```

Comment:
Doesn't this move the cache from a file cache into an in-memory cache? That would be undesired.

Reply:
Also, I am not sure I like having the cache interface split out into specific caches; the way it is done today all caches related to the repository is stored under the same top-level dir iirc. We will want to retain this.

Reply:
Currently the match and package caches are actually both in-memory, see [this line.](https://github.com/python-poetry/poetry/blob/master/src/poetry/repositories/cached.py#L36). The only in-file cache is for release metadata.

I could definitely combine these all back together, but I was wondering if, since _match_cache and _package_cache are already in-memory and only used in these subclasses, perhaps they should stay split? Unless there are plans for the cache to make use of all three stores in the future.

Reply:
I'm pretty sure that the `packages` cache is more or less worthless: the request to the server has `cachecontrol` taking care of caching already, at best the extra layer is saving us re-parsing the JSON in the response.

The 'matches' cache is similar, though perhaps it saves _slightly_ more work.

git history shows that `cachy` was introduced first, and `cachecontrol` followed later.  The caching provided by `cachecontrol` makes these two in-memory caches near-enough redundant I reckon.

Still, I'd be tempted to get rid of them both altogether for the simplification - unless testing shows that they really make a meaningful difference.

Reply:
I could definitely go ahead and do a profile to compare these two. This would really help reduce the size of the caching module overall, as we could then compress most of this logic into a single class.